### ISSUE TITLE
Incorrect error on method name conflict with its type parameter

### ISSUE THREAD

**Issue by gafter** [dotnet/roslyn#7506]
The following code is allowed by the language spec and allowed by the mono compiler. For compat Roslyn should allow it too.

```cs
void T<T>()  // Should be no error
```

**Comment by orthoxerox** [comment]
Is it possible to relax the following restriction in section 3.3 of the spec?

> Except for overloaded instance constructor declarations and static constructor declarations, a class or struct cannot contain a member declaration with the same name as the class or struct.

**Comment by gafter** [comment]
@orthoxerox Possible, I believe, but error-prone (too easy to confuse with constructors) and therefore unlikely.

**Comment by orthoxerox** [comment]
@gafter What about allowing it for properties/fields only? They are unlikely to be confused with constructors.

**Comment by gafter** [comment]
@orthoxerox This is not the best place to propose language changes. This issue is about making the compiler obey the existing spec and be compatible with other compilers.

**Comment by gafter** [comment]
Note to self: see https://github.com/dotnet/roslyn/issues/7507#issuecomment-164898986 for @SLaks suggested fix.
